### PR TITLE
Remove ModuleFinder.findModuleInClasspath

### LINKED ISSUE #330: Remove ModuleFinder.findModuleInClasspath
Split form #320 

https://github.com/morganey-lang/Morganey/blob/458946d8fefc71fffc8c98834a7752c9081925ab/src/main/scala/me/rexim/morganey/module/ModuleFinder.scala#L15

All of the code that depends on this function should be refactored accordingly. Getting rid of this function essentially eliminates any violations of Module Entity encapsulation.

The only code that uses this function is MorganeyCompiler.

### LINKED ISSUE #331: Extract mockClassLoaderResource into reusable thing
https://github.com/morganey-lang/Morganey/blob/458946d8fefc71fffc8c98834a7752c9081925ab/src/test/scala/me/rexim/morganey/module/ModuleSpec.scala#L17

Reuse that thing in ModuleFinderSpec

### PR DESCRIPTION
**Pull request by rexim**
Close #330 
Close #331

### CODE REVIEW

File: funtests/src/test/scala/me/rexim/morganey/funtests/MorganeyProcess.scala [hunk 1]
```diff
@@ -5,5 +5,5 @@ import me.rexim.morganey.BuildInfo
 
 trait MorganeyProcess {
   def morganey(args: String*): ProcessBuilder =
-    s"java -jar ./target/scala-2.11/morganey-assembly-${BuildInfo.version}.jar ${args.mkString(" ")}"
+    s"java -cp .:./target/scala-2.11/morganey-assembly-${BuildInfo.version}.jar me.rexim.morganey.Main ${args.mkString(" ")}"
```
Comment (rexim) [inline funtests/src/test/scala/me/rexim/morganey/funtests/MorganeyProcess.scala]: That should be separated by `;` on Windows

File: src/main/scala/me/rexim/morganey/module/Module.scala [hunk 2]
```diff
@@ -11,24 +12,35 @@ class Module(modulePath: ModulePath, classLoader: ClassLoader = Module.getClass.
...
+    } else {
+      Success(Nil)
+    }
+
   private def nodes: Try[List[MorganeyNode]] = {
     val CanonicalPath(canonicalPath) = modulePath.asCanonicalPath
     val ResourcePath(resourcePath) = modulePath.asResourcePath
 
-    lazy val moduleNotFound = Failure(new ModuleNotFoundException(s"$canonicalPath module was not found"))
+    lazy val moduleNotFound = Failure(new ModuleNotFoundException(s"$resourcePath module was not found"))
```
Comment (rexim) [inline src/main/scala/me/rexim/morganey/module/Module.scala]: Revert that back to `canonicalPath`. It's a left-over from troubleshooting.

File: funtests/src/test/scala/me/rexim/morganey/funtests/MorganeyProcess.scala (hunk 1 above)
Comment (rexim) [reply funtests/src/test/scala/me/rexim/morganey/funtests/MorganeyProcess.scala]: Also, starting Morganey programs like that is just unusable. Create an issue for UX improvement.

Comment (ForNeVeR) [reply funtests/src/test/scala/me/rexim/morganey/funtests/MorganeyProcess.scala]: > That should be separated by `;` on Windows

Note there's `File.pathSeparator` for that. It resolves to either `:` or `;` depending on the current platform.

**Review by ForNeVeR — APPROVED** [review]

**Review by keddelzz — APPROVED** [review]

File: src/test/scala/me/rexim/morganey/interpreter/MorganeyCompilerSpec.scala [hunk 3]
```diff
@@ -27,21 +25,16 @@ class MorganeyCompilerSpec extends FlatSpec with Matchers {
...
 
-  "Compiler" should "not interpret empty loading nodes and just ignore them" in {
-    MorganeyCompiler.interpretNode(MorganeyLoading(None), moduleFinder, Set()) should be (Success(List()))
-  }
-
-  "Compiler" should "should returns bindings as is during their interpretation as a node" in {
-    val binding = MorganeyBinding(lvar("khooy"), lvar("khooy"))
-    MorganeyCompiler.interpretNode(binding, moduleFinder, Set()) should be (Success(List(binding)))
-  }
+  it should ""
```
Comment (keddelzz) [inline src/test/scala/me/rexim/morganey/interpreter/MorganeyCompilerSpec.scala]: Is this line intended to exist or is that a leftover from refactoring?

Comment (rexim) [reply src/test/scala/me/rexim/morganey/interpreter/MorganeyCompilerSpec.scala]: Haha! Nice one! :D Yeah, I think it's a leftover of some sort. Thanks for pointing that out. I'll remove it.

**Comment by rexim** [comment]
@ForNeVeR @keddelzz thank you guys for the review!

**Pull request merged**